### ISSUE TITLE
RFE: use dnf repoquery dependency data, not spec file

### ISSUE THREAD

**Issue by michel-slm** [juhp/rpmbuild-order#3]
`fbrnch` seems to be parsing spec files to determine dependencies between packages. This does not work for packaging ecosystems that dynamically compute build requirements during the build -- Python SIG is [recommending that](https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#Automatically-generated-dependencies) and [all our Rust packages use dynamic BRs](https://docs.fedoraproject.org/en-US/packaging-guidelines/Rust/#_automatic_dependency_generation)

Right now, this doesn't work, e.g.

```
$ sudo dnf repoquery --disablerepo=\* --enablerepo=rawhide-source --requires rust-below | grep below
Last metadata expiration check: 0:17:52 ago on Wed 05 Jan 2022 10:51:22 AM PST.
(crate(below-common/default) >= 0.4.1 with crate(below-common/default) < 0.5.0~)
(crate(below-config/default) >= 0.4.1 with crate(below-config/default) < 0.5.0~)
(crate(below-dump/default) >= 0.4.1 with crate(below-dump/default) < 0.5.0~)
(crate(below-model/default) >= 0.4.1 with crate(below-model/default) < 0.5.0~)
(crate(below-store/default) >= 0.4.1 with crate(below-store/default) < 0.5.0~)
(crate(below-view/default) >= 0.4.1 with crate(below-view/default) < 0.5.0~)
```

`fbrnch parallel -n` claims these can all be built in parallel though:
```
❯ fbrnch parallel -n f34 rust-below rust-below-common rust-below-config rust-below-dump rust-below-model rust-below-store rust-below-view                                        
                                                                                                                                                                                 
Building parallel layer of 7 packages:                                                                                                                                           
rust-below rust-below-common rust-below-config rust-below-dump rust-below-model rust-below-store rust-below-view                                                                 
(0 more layer left) 
...
```

**Comment by juhp** [comment]
I meant to note that as a first step I had implemented https://github.com/juhp/fbrnch/commit/9df2aea9f1de93f3ee5ab5eebeff9e1426237ecd to handle pulling in dynamic BRs for local fbrnch builds

**Comment by juhp** [comment]
Okay I have done an initial implementation of Dynamic BuildRequires support in rpmbuild-order.
Would you like to try to test it before I do the next release?

It seems to work for me for your "below" crates at least and also lightly tested on a couple of python packages.
(On the other hand it looks hard to hack support for golang packages.)

I may still look at adding a --repoquery option later - though it will be very slow for large package sets.

**Comment by juhp** [comment]
This is how i tested:
```shellsession
$ rpmbuild-order sort rust-*
rust-chrono rust-below-model rust-below-config rust-below-common rust-below-store rust-below-view rust-below-dump rust-below

$ rpmbuild-order sort python-copr/ python-mock/
python-mock/ python-copr/
```

**Comment by juhp** [comment]
I am hoping to get this out soon - I was waiting to do some refactor to the new logic to determine dynamic Provides from source files to support golang correctly.

**Comment by juhp** [comment]
I believe I now finally got round to handling golang provides too. :-)

**Comment by juhp** [comment]
Released now in 0.4.6, but I would welcome testing and feedback.
Should get into fedora soon.

**Comment by juhp** [comment]
It is in updates-testing now, along with fbrnch-1.1.1
